providers/infrastructure-aws/v2.0.0-beta.1/cconly/base.yaml

```diff
@@ -1015,22 +1019,22 @@ spec:
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
 kind: AWSClusterTemplate
 metadata:
-  name: aws-cluster-template
+  name: #@ "tkg-aws-default-{}-cluster".format(data.values.AWS_CLUSTER_CLASS_VERSION)
   namespace: #@ data.values.NAMESPACE
 spec:
   template:
     spec:
-      region: "${AWS_REGION}"
-      sshKeyName: "${AWS_SSH_KEY_NAME}"
+      region: #@ data.values.AWS_REGION
```

Comment:
I had a confusion for a long time about the difference between environment variable (by export) and config variable (in config file), when should we use environment variable and when should use config variable, or they are equivalent through some cli internal Golang code (uncertain...)? Thus, does this kind of change break user behavior?

Reply:
I remember a (uncertain) time ago when deleting AWS management cluster, AWS_REGION is necessary to be present as environment, not sure what's the current status.

Reply:
Environment variable can not be interpolated by our ytt processor, you will see them still as ${AWS_REGION} after it is rendered